src/evm/Readme.md

```diff
@@ -2,73 +2,72 @@
 
-`AccountInfo` contains basic account information such as balance, nonce and code (in case of smart contract accounts).
+```
+address: Address,
+account: AccountInfo,
+permanent_storage: Option<HashMap<U256, U256>>,
+mocked: bool,
+```
 
-Permanent storage lets us set some storage slots. If a value of a certain storage slot is needed during simulation, the
-state will first check the permanent storage set here. Only if a slot is not set in the permanent storage, the state
-will query an Ethereum node (but see the next section).
+- `AccountInfo`: Contains core account details like balance, nonce, and code (for smart contract accounts).
+- `permanent_storage`: Allows setting predefined storage slots. When the simulation requires a specific storage slot,
+  the engine first checks this storage.
+- `mocked`: Determines whether the account is mocked (see below).
 
-The next section describes the `mocked` argument.
-
-### Mocked and non-mocked accounts
-
-An overriden account may be set as mocked or non-mocked.
-
-For non-mocked accounts (`mocked=false`), missing storage slots will be queried from a node.
+### Simulation-time overrides
 
-For mocked accounts (`mocked=true`), nothing will be queried from a node. Missing slots have value of 0: if a certain
-storage slot is needed during simulation, but has neither been set in the account's permanent storage (see the previous
-section) nor in the simulation-time overrides (see the next section), then the value of such slot will be 0.
+You can override the chain state only for the duration of a single simulation. This is done by
+setting `SimulationParameters.overrides` to a mapping of form `HashMap<Address, HashMap<U256, U256>>`.
 
-If you are overriding contract code, you most probably want to set the account to be mocked. That's because overridden
-code might use different storage slots than the code on-chain, so querying a node for storage of such account wouldn't
-make sense.
+Overrides set here take precedence over an account's permanent storage.
 
-### Simulation-time overrides
+### SimulationDB
 
-You can override the chain state only for the duration of a single simulation. This is done by
-setting `SimulationParameters.overrides` to a mapping of form `address -> storage slot index -> storage slot value`.
+#### Mocked vs. Non-Mocked Accounts
 
-Overrides set this way will take priority over accounts' permanent storage and querying from a node.
+- Non-Mocked Accounts (`mocked=false`):
+    - Missing storage slots are queried from the Ethereum node.
+- Mocked Accounts (`mocked=true`):
+    - No data is queried from the node.
+    - Missing storage slots default to 0 if they are not set in either permanent storage or simulation-time overrides.
 
-## See also
+Mocking is especially useful when overriding contract code, as the overridden code might rely on different storage slots
+than the on-chain version. Querying a node for such storage would not yield meaningful results.
```

Comment:
What does this actually mean, in non-diplomatic terms? haha. Can you give an example?

Reply:
I just re wrote what it said there 😂  but I think it's like you can mock the storage slots to be wtv you want. This is particularly useful if you are overriding the contract code to be different than on chain code, because the storage slots might be different

Reply:
I like what you just said a lot better

Reply:
I agree - this can be worded better.